internal/asm/f64/ger_amd64.s

```diff
@@ -0,0 +1,603 @@
+	ADDPD  X8, X6            \
+	MOVUPS X5, (A_PTR)       \
+	MOVUPS X6, 2*SIZE(A_PTR) \
+	ADDQ   $4*SIZE, A_PTR
+
+#define KERNEL_1x2 \
+	MULPD X1, X5
+
+#define STORE_1x2 \
+	MOVUPS (A_PTR), X6    \
+	ADDPD  X6, X5         \
+	MOVUPS X5, (A_PTR)    \
+	ADDQ   $2*SIZE, A_PTR
+
+#define KERNEL_1x1 \
+	MOVSD (Y_PTR), X5 \
+	MULSD X1, X5
+
+#define STORE_1x1 \
+	ADDSD (A_PTR), X5  \
+	MOVSD X5, (A_PTR)  \
+	ADDQ  $SIZE, A_PTR
+
+// func Ger(m, n uintptr, alpha float64,
+//	x []float64, incX uintptr,
+//	y []float64, incY uintptr,
+//	a []float64, lda uintptr)
+TEXT ·Ger(SB), NOSPLIT, $0
+	MOVQ M_DIM, M
+	MOVQ N_DIM, N
+	CMPQ M, $0
+	JE   end
+	CMPQ N, $0
+	JE   end
+
+	MOVDDUP alpha+16(FP), X0
+
+	MOVQ x_base+24(FP), X_PTR
+	MOVQ y_base+56(FP), Y_PTR
+	MOVQ a_base+88(FP), A_ROW
+	MOVQ incX+48(FP), INC_X       // INC_X = incX * sizeof(float64)
+	SHLQ $3, INC_X
+	MOVQ lda+112(FP), LDA         // LDA = LDA * sizeof(float64)
+	SHLQ $3, LDA
+	LEAQ (LDA)(LDA*2), LDA3       // LDA3 = LDA * 3
+	LEAQ (INC_X)(INC_X*2), INC3_X // INC3_X = INC_X * 3
+	MOVQ A_ROW, A_PTR
+
+	XORQ    TMP2, TMP2
+	MOVQ    M, TMP1
+	SUBQ    $1, TMP1
+	IMULQ   INC_X, TMP1
+	NEGQ    TMP1
+	CMPQ    INC_X, $0
+	CMOVQLT TMP1, TMP2
+	LEAQ    (X_PTR)(TMP2*SIZE), X_PTR
+
+	CMPQ incY+80(FP), $1 // Check for dense vector Y (fast-path)
+	JG   inc
+	JL   end
+
+	SHRQ $2, M
+	JZ   r2
+
+r4:
+	// LOAD 4
+	LOAD4
+
+	MOVQ N_DIM, N
+	SHRQ $2, N
+	JZ   r4c2
+
+r4c4:
+	// 4x4 KERNEL
+	KERNEL_LOAD4
+	KERNEL_4x4
+	STORE_4x4
+
+	ADDQ $4*SIZE, Y_PTR
+
+	DECQ N
+	JNZ  r4c4
+
+	// Reload ALPHA after it's clobbered by STORE_4x4
+	MOVDDUP alpha+16(FP), ALPHA
+
+r4c2:
+	TESTQ $2, N_DIM
+	JZ    r4c1
+
+	// 4x2 KERNEL
+	KERNEL_LOAD2
+	KERNEL_4x2
+	STORE_4x2
+
+	ADDQ $2*SIZE, Y_PTR
+
+r4c1:
+	TESTQ $1, N_DIM
+	JZ    r4end
+
+	// 4x1 KERNEL
+	KERNEL_4x1
+	STORE_4x1
+
+	ADDQ $SIZE, Y_PTR
+
+r4end:
+	LEAQ (X_PTR)(INC_X*4), X_PTR
+	MOVQ Y, Y_PTR
+	LEAQ (A_ROW)(LDA*4), A_ROW
+	MOVQ A_ROW, A_PTR
+
+	DECQ M
+	JNZ  r4
+
+r2:
+	TESTQ $2, M_DIM
+	JZ    r1
+
+	// LOAD 2
+	LOAD2
+
+	MOVQ N_DIM, N
+	SHRQ $2, N
+	JZ   r2c2
+
+r2c4:
+	// 2x4 KERNEL
+	KERNEL_LOAD4
+	KERNEL_2x4
+	STORE_2x4
+
+	ADDQ $4*SIZE, Y_PTR
+
+	DECQ N
+	JNZ  r2c4
+
+r2c2:
+	TESTQ $2, N_DIM
+	JZ    r2c1
+
+	// 2x2 KERNEL
+	KERNEL_LOAD2
+	KERNEL_2x2
+	STORE_2x2
+
+	ADDQ $2*SIZE, Y_PTR
+
+r2c1:
+	TESTQ $1, N_DIM
+	JZ    r2end
+
+	// 2x1 KERNEL
+	KERNEL_2x1
+	STORE_2x1
+
+	ADDQ $SIZE, Y_PTR
+
+r2end:
+	LEAQ (X_PTR)(INC_X*2), X_PTR
+	MOVQ Y, Y_PTR
+	LEAQ (A_ROW)(LDA*2), A_ROW
+	MOVQ A_ROW, A_PTR
+
+r1:
+	TESTQ $1, M_DIM
+	JZ    end
+
+	// LOAD 1
+	LOAD1
+
+	MOVQ N_DIM, N
+	SHRQ $2, N
+	JZ   r1c2
+
+r1c4:
+	// 1x4 KERNEL
+	KERNEL_LOAD4
+	KERNEL_1x4
+	STORE_1x4
+
+	ADDQ $4*SIZE, Y_PTR
+
+	DECQ N
+	JNZ  r1c4
+
+r1c2:
+	TESTQ $2, N_DIM
+	JZ    r1c1
+
+	// 1x2 KERNEL
+	KERNEL_LOAD2
+	KERNEL_1x2
+	STORE_1x2
+
+	ADDQ $2*SIZE, Y_PTR
+
+r1c1:
+	TESTQ $1, N_DIM
+	JZ    end
+
+	// 1x1 KERNEL
+	KERNEL_1x1
+	STORE_1x1
+
+	ADDQ $SIZE, Y_PTR
+
+r1end:
```

Comment:
Is this part necessary? Also in the `inc` case.

Reply:
No, both blocks are dead code.  Deleted